src/svc/bigquery.rs

```diff
@@ -0,0 +1,455 @@
+    pub name: String,
+    pub mode: String,
+
+    #[serde(rename="type")]
+    pub type0: String,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub fields: Option<Vec<TableField>>,
+}
+
+#[derive(Serialize, Debug)]
+pub struct SubmitQueryRequest {
+    pub kind: String,
+    pub query: String,
+
+    #[serde(rename="timeoutMs")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeout_ms: Option<usize>,
+
+    #[serde(rename="useLegacySql")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub use_legacy_sql: Option<bool>,
+
+    #[serde(rename="useQueryCache")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub use_query_cache: Option<bool>,
+
+    #[serde(rename="dryRun")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub dry_run: Option<bool>,
+
+    #[serde(rename="maxResults")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_results: Option<usize>,
+}
+
+impl Default for SubmitQueryRequest {
+    fn default() -> Self {
+        SubmitQueryRequest {
+            kind: QUERY_RESOURCE_KIND.to_string(),
+            query: Default::default(),
+            timeout_ms: None,
+            use_legacy_sql: None,
+            use_query_cache: None,
+            dry_run: None,
+            max_results: None,
+        }
+    }
+}
+
+#[derive(Deserialize, Default, Debug)]
+pub struct SubmitQueryResponse {
+    pub schema: Option<TableFieldSchema>,
+
+    #[serde(rename="jobReference")]
+    pub job_reference: JobReference,
+
+    #[serde(rename="totalRows")]
+    pub total_rows: Option<String>,
+
+    pub rows: Option<Vec<TableRow>>,
+
+    #[serde(rename="jobComplete")]
+    pub job_complete: bool,
+
+    #[serde(rename="pageToken")]
+    pub page_token: Option<String>,
+
+    pub errors: Option<Vec<QueryError>>,
+}
+
+#[derive(Deserialize, Default, Debug)]
+pub struct JobReference {
+    #[serde(rename="projectId")]
+    pub project_id: String,
+
+    #[serde(rename="jobId")]
+    pub job_id: String,
+}
+
+#[derive(Deserialize, Default, Debug)]
+pub struct QueryError {
+    pub reason: Option<String>,
+    pub location: Option<String>,
+    pub message: Option<String>,
+
+    #[serde(rename="debugInfo")]
+    pub debug_info: Option<String>,
+}
+
+#[derive(Deserialize, Default, Debug)]
+pub struct CancelQueryResponse {}
+
+#[derive(Serialize, Default, Debug)]
+pub struct AwaitQueryRequest {
+    #[serde(rename="timeoutMs")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub timeout_ms: Option<usize>,
+
+    #[serde(rename="maxResults")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_results: Option<usize>,
+
+    #[serde(rename="startIndex")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub start_index: Option<usize>,
+
+    #[serde(rename="pageToken")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub page_token: Option<String>,
+}
+
+impl AwaitQueryRequest {
+    fn to_query(&self) -> String {
```

Comment:
it seems like maybe we could make this work with serde? not critical, but would be nice